### ISSUE TITLE
Fatal error: out of memory.

### ISSUE THREAD

**Issue by jflowers1974** [MinaProtocol/mina#6851]
Machine:  'bare metal' 8 cores 48 GB ram Ubuntu 18

Been able to keep the system running (block producer only) for 6 hrs or so.  Updated the software/peers.txt earlier today Nov 24 and was up and running for ~5 hrs without issue (sync'ed, good number of peers).  Went grocery shopping and came home to this:
...
2020-11-25 00:39:22 UTC [Warn] failed to reset stream (this means it was probably closed successfully): $error
	error: {
  "string":
    "RPC #15870 failed: \"internal RPC error error: unknown stream_idx\""
}
2020-11-25 00:39:22 UTC [Info] verifier successfully got sigkill
2020-11-25 00:39:22 UTC [Info] prover successfully stopped
2020-11-25 00:39:24 UTC [Info] Sending transition_chain_proof to $peer
	peer: {
  "host": "94.103.87.3",
  "peer_id": "12D3KooWLPLqV1aYPG2fEVPY1PXo3vBGW6t65ZnYh1yAU4WvkJVL",
  "libp2p_port": 8302
}
2020-11-25 00:39:25 UTC [Info] Daemon started process of kind "Verifier" with pid 19896
2020-11-25 00:39:54 UTC [Error] Validation error: external transition with state hash $state_hash was rejected for reason "invalid time"
	state_hash: "3NLCsUFSnRPoQZTaoBNJHNrZ8jL5Fj3Q1DVXUZiugkRadWbBoEia"
2020-11-25 00:40:04 UTC [Warn] failed to reset stream (this means it was probably closed successfully): $error
	error: {
  "string":
    "RPC #16412 failed: \"internal RPC error error: unknown stream_idx\""
}
Fatal error: out of memory.

Again, I've got 48 GB ram for this locally running VM (the base system has 64 GB).

**Comment by bkase** [comment]
@jflowers1974 was this with the new build (that we sent out yesterday) or the old one?

**Comment by gregbostrom** [comment]
I've had three of these today
```
log.2020-11-27_07:13:17:Fatal error: out of memory.
log.2020-11-27_14:51:47:Fatal error: out of memory.
log.2020-11-27_17:43:03:Fatal error: out of memory.
```
With version:
```
[DIRTY]7f0440c609bfd1e2bc4130f06d09f72763c25b67
```
I have 64GB RAM on this VPS.  Here is what it looks like without coda running.
```
$ free -mh
              total        used        free      shared  buff/cache   available
Mem:            61G        227M         53G         79M        7.3G         60G
Swap:            0B          0B          0B
```
If I watch memory usage while coda is running, I'm able to see the memory usage steadily grow until there is none available.
```
watch -n3 -d free -mh
```
So at the moment the problem is very reproducible.  It wasn't happening yesterday.

**Comment by gregbostrom** [comment]
Curious, this memory problem is not happening on my system today.

**Comment by EmreNOP** [comment]
check this one

https://github.com/MinaProtocol/mina/issues/6965

**Comment by hsq125** [comment]
I think the problem of memory allocation does still exist in version `1.2.0beta1-c856692-mainnet`

Possibly more hidden than before

`2021-07-02 10:28:28 UTC [Error] Possible reason for signal: "Process killed because out of memory"`

**Log extract**
2021-07-02 10:27:31 UTC [Warn] RPC call error for "get_transition_chain_proof"
2021-07-02 10:27:35 UTC [Error] error sending message on stream 25703: $error
	error: {
  "commit_id": "c856692fddc525a673ba075f714811b5c50bd3a7",
  "string":
    "RPC #46101 failed: \"only wrote 0 out of 9 bytes error: libp2p error error: closed stream\""
}
2021-07-02 10:27:35 UTC [Warn] RPC call error for "get_transition_chain"
2021-07-02 10:27:41 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] Not rebroadcasting block $state_hash because it was received "1 slots too late"
	state_hash: "3NLex6ZoBK5FxhxRwPwmEBHLVEycVBf2kWULC9gTcDwPq5wskHQz"
2021-07-02 10:27:51 UTC [Info] Saw block with state hash $state_hash
	state_hash: "3NLex6ZoBK5FxhxRwPwmEBHLVEycVBf2kWULC9gTcDwPq5wskHQz"
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:51 UTC [Warn] initial_validate: disconnected chain
2021-07-02 10:27:54 UTC [Warn] RPC call error for "get_transition_chain_proof"
2021-07-02 10:27:55 UTC [Info] Received a block from $sender
	sender: {
  "Remote": {
    "host": "34.122.226.192",
    "peer_id": "12D3KooWAbpbSc9WbfkrJE8FQNLebzL9A7WGUVaMFaEjexW4MPmU",
    "libp2p_port": 8302
  }
}
2021-07-02 10:28:05 UTC [Fatal] Unhandled top-level exception: $exn
Generating crash report
	exn: {
  "commit_id": "c856692fddc525a673ba075f714811b5c50bd3a7",
  "sexp": [
    "monitor.ml.Error",
    "Cached item has already been finalized",
    [
      "Raised at file \"src/error.ml\" (inlined), line 9, characters 14-30",
      "Called from file \"src/lib/cache_lib/impl.ml\", line 173, characters 30-51",
      "Called from file \"src/lib/cache_lib/impl.ml\", line 199, characters 6-69",
      "Called from file \"src/lib/ledger_catchup/super_catchup.ml\", line 928, characters 25-58",
      "Called from file \"src/deferred1.ml\", line 17, characters 40-45",
      "Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5",
      "Called from file \"src/job_queue.ml\", line 171, characters 6-47",
      "Caught by monitor coda"
    ]
  ],
  "backtrace": [
    "Raised at file \"format.ml\" (inlined), line 242, characters 35-52",
    "Called from file \"format.ml\", line 469, characters 8-33",
    "Called from file \"format.ml\", line 484, characters 6-24"
  ]
}
2021-07-02 10:28:05 UTC [Info] Updating new available work took 21.804094314575195 ms
2021-07-02 10:28:28 UTC [Error] Daemon child process 97 terminated after receiving signal "sigkill"
2021-07-02 10:28:28 UTC [Error] Possible reason for signal: "Process killed because out of memory"
2021-07-02 10:28:28 UTC [Error] Child process of kind "Prover" with pid 97 has unexpectedly terminated
2021-07-02 10:28:28 UTC [Fatal] Unhandled top-level exception: $exn
Generating crash report
	exn: {
  "commit_id": "c856692fddc525a673ba075f714811b5c50bd3a7",
  "sexp": [
    "monitor.ml.Error",
    [ "Failure", "Child process of kind Prover has unexpectedly terminated" ],
    [
      "Raised at file \"stdlib.ml\", line 33, characters 17-33",
      "Called from file \"src/app/cli/src/cli_entrypoint/mina_cli_entrypoint.ml\", line 579, characters 10-94",
      "Called from file \"src/deferred0.ml\", line 56, characters 64-69",
      "Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5",
      "Called from file \"src/job_queue.ml\", line 171, characters 6-47",
      "Caught by monitor coda"
    ]
  ],
  "backtrace": [
    "Raised by primitive operation at file \"src/signal.ml\", line 162, characters 6-61"
  ]
}
2021-07-02 10:28:28 UTC [Error] verifier terminated unexpectedly
2021-07-02 10:28:28 UTC [Info] Starting a new verifier process
2021-07-02 10:28:28 UTC [Info] verifier successfully stopped
2021-07-02 10:28:28 UTC [Info] Rebroadcasting $state_hash
	state_hash: "3NLnPyR37jEecT8RdmJ6m7HKnNxicr8gj1kKA4x5iZC4nueGevJ4"
2021-07-02 10:28:28 UTC [Fatal] libp2p_helper process died unexpectedly: "died after receiving sigkill (signal number 9)"
2021-07-02 10:28:28 UTC [Error] error during validationComplete, ignoring and continuing: $error
	error: {
  "commit_id": "c856692fddc525a673ba075f714811b5c50bd3a7",
  "string":
    "helper process already exited (doing RPC {\"seqno\":25735,\"is_valid\":\"accept\"})"
}

**Comment by shimkiv** [comment]
Closing this one because of no activity for a long time and I hope that the most recent versions of Mina Daemon behave more stable for you.
Please feel free to report new issues should you face any.
Thank you.
